ext/opentelemetry-exporter-cloud-monitoring/src/opentelemetry/exporter/cloud_monitoring/__init__.py

```diff
@@ -0,0 +1,136 @@
+        :return:
+        """
+        # TODO: Leverage this better
+
+    def _batch_write(self, series: TimeSeries):
+        """ Cloud Monitoring allows writing up to 200 time series at once
+
+        :param series: ProtoBuf TimeSeries
+        :return:
+        """
+        write_ind = 0
+        while write_ind < len(series):
+            self.client.create_time_series(
+                self.project_name,
+                series[write_ind : write_ind + MAX_BATCH_WRITE],
+            )
+            write_ind += MAX_BATCH_WRITE
+
+    def _get_metric_descriptor(self, record: MetricRecord) -> MetricDescriptor:
+        """ We can map Metric to MetricDescriptor using Metric.name or
+        MetricDescriptor.type. We create the MetricDescriptor if it doesn't
+        exist already and cache it. Note that recreating MetricDescriptors is
+        a no-op if it already exists.
+
+        :param record:
+        :return:
+        """
+        descriptor_type = "custom.googleapis.com/OpenTelemetry/{}".format(
+            record.metric.name
+        )
+        if descriptor_type in self._metric_descriptors:
+            return self._metric_descriptors[descriptor_type]
+        descriptor = {
+            "name": "display_name",
+            "type": descriptor_type,
+            "display_name": record.metric.name,
+            "description": record.metric.description,
+            "labels": [],
+        }
+        for key, value in record.labels:
+            if isinstance(value, str):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="STRING")
+                )
+            elif isinstance(value, int):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="INT64")
+                )
+            elif isinstance(value, bool):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="BOOL")
+                )
+        if isinstance(record.aggregator, CounterAggregator):
+            descriptor["metric_kind"] = MetricDescriptor.MetricKind.GAUGE
+        else:
+            logger.warning(
+                "Unsupported aggregation type %s, ignoring it",
+                type(record.aggregator).__name__,
+            )
+            return None
+        if record.metric.value_type == int:
+            descriptor["value_type"] = MetricDescriptor.ValueType.INT64
+        elif record.metric.value_type == float:
+            descriptor["value_type"] = MetricDescriptor.ValueType.DOUBLE
+        descriptor = self.client.create_metric_descriptor(
+            self.project_name, MetricDescriptor(**descriptor)
+        )
+        self._metric_descriptors[descriptor_type] = descriptor
+        return descriptor
+
+    def export(
+        self, metric_records: Sequence[MetricRecord]
+    ) -> "MetricsExportResult":
+        all_series = []
+        for record in metric_records:
```

Comment:
would you sometimes want more than one point in a timeseries? what if we want to wait for a number of timeseries points before exporting to do bigger batches, so perhaps there are 100 MetricRecords that really belong to just two timeseries? can this happen? and if so could you make this just call createTimeSeries once per series?

Reply:
actually it looks like you can only write one point at a time, so never mind! https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.timeSeries/create